src/elfpy/pricing_models.py

```diff
@@ -849,37 +1040,41 @@ def calc_out_given_in(
         """
         assert in_ > 0, f"pricing_models.calc_out_given_in: ERROR: expected in_ > 0, not {in_}!"
         assert (
-            share_reserves > 0
-        ), f"pricing_models.calc_out_given_in: ERROR: expected share_reserves > 0, not {share_reserves}!"
+            share_reserves >= 0
+        ), f"pricing_models.calc_out_given_in: ERROR: expected share_reserves >= 0, not {share_reserves}!"
         assert (
-            bond_reserves > 0
-        ), f"pricing_models.calc_out_given_in: ERROR: expected bond_reserves > 0, not {bond_reserves}!"
+            bond_reserves >= 0
+        ), f"pricing_models.calc_out_given_in: ERROR: expected bond_reserves >= 0, not {bond_reserves}!"
         assert (
             1 >= fee_percent >= 0
         ), f"pricing_models.calc_out_given_in: ERROR: expected 1 >= fee_percent >= 0, not {fee_percent}!"
         assert (
             1 > time_remaining >= 0
         ), f"pricing_models.calc_out_given_in: ERROR: expected 1 > time_remaining >= 0, not {time_remaining}!"
         assert share_price >= init_share_price >= 1, (
-            f"pricing_models.calc_out_given_in: ERROR: expected share_price >= init_share_price >= 1, "
-            f"not share_price={share_price} and init_share_price={init_share_price}!"
+            "pricing_models.calc_out_given_in: ERROR: expected share_price >= init_share_price >= 1, not"
+            f" share_price={share_price} and init_share_price={init_share_price}!"
         )
+
         # TODO: Break this function up to use private class functions
         # pylint: disable=too-many-locals
         scale = share_price / init_share_price
         time_elapsed = 1 - time_remaining
         total_reserves = share_price * share_reserves + bond_reserves
         spot_price = self.calc_spot_price_from_reserves(
-            share_reserves, bond_reserves, init_share_price, share_price, time_remaining
+            share_reserves=share_reserves,
+            bond_reserves=bond_reserves,
+            init_share_price=init_share_price,
+            share_price=share_price,
+            time_remaining=time_remaining
         )
         # We precompute the YieldSpace constant k using the current reserves and
         # share price:
         #
-        # k = (c / μ) * (μ * z)**(1 - τ) + (2y + cz)**(1 - τ)
         k = price_utils.calc_k_const(share_reserves, bond_reserves, share_price, init_share_price, time_elapsed)
         if token_out == "base":
-            d_bonds = in_
-            in_reserves = bond_reserves + total_reserves
+            d_bonds = in_  # PTs passed in
+            in_reserves = bond_reserves + total_reserves  # add virtual liquidity
```

Comment:
If we're going to add these comments, let's add them in all if-clauses in the pricing model with this characteristic.